apps/teams/utils.py

```diff
@@ -32,9 +32,11 @@ def set_current_team(team) -> Token:
         get_current_team(team)
     ```
     """
-    if existing_team := get_current_team():
+    team = _unwrap_lazy(team)
+    existing_team = get_current_team()
```

Comment:
The existing team can be a lazy object as well. See [here](https://dimagi.sentry.io/issues/6545279442/?project=4505001320316928&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&sort=date&stream_index=1)

Reply:
I can only be a lazy object if it's set to a lazy object and since we're now unwrapping lazy objects before we set them I don't think that's an issue.

Reply:
Ah of course, you're right. I was using a single brain cell